hm_pyhelper/diagnostics/diagnostics_report.py

```diff
@@ -33,10 +33,33 @@ def __init__(self, diagnostics=[], **kwargs):
 
         if DIAGNOSTICS_ERRORS_KEY not in self:
             self.__setitem__(DIAGNOSTICS_ERRORS_KEY, [])
+
         self.diagnostics = diagnostics
 
+        # Needed for more verbose messages at HPT end. This attribute gets
+        # queried to display exactly which keys failed causing the diagnostics
+        # to fail.
+        self.manufacturing_errors = []
+
     def passed(self):
```

Comment:
I think this method should be deprecated/removed. I would not add this new logic to it. This is because we are moving away from supporting a top-level PF/diagnostics_passed. So let's create a new method instead of modifying this one.

That new method should accept a list of `keys_to_check` as a parameter rather than it being hardcoded. That way we can use this method with different arguments from different contexts.

Reply:
Then it would be up to the calling context to specifies which keys to check? That doesn't look like good encapsulation logic. Perhaps if we have a few well defined scenarios (like all errors, manufacturing errors) we should define methods for those like (has_errors or errors, has_manufacturing_errors etc) and these methods check for a specific set of keys instead of asking the caller to specify the keys?

If the caller has to specify the keys then Miner.did_miner_pass_diagnostics method was already doing that and adding this method does not provide much benefit.

`keys_to_check` can be for all errors or manufacturing errors specified within the DiagnosticsReport class itself. And we can have 2 
`@property` methods named `has_errors` and `has_manufacturing_errors` that use 2 different set of keys. This way the logic is still all kept inside diagnostics report itself.